### PR TITLE
Add deferrable flag and optionally enter deferred state

### PR DESCRIPTION
**Pull request by jlaneve**

### CODE REVIEW

File: fivetran_provider_async/operators.py [hunk 1]
```diff
@@ -67,18 +72,19 @@ def execute(self, context: Dict[str, Any]) -> None:
...
+                timeout=self.execution_timeout,
+                trigger=FivetranTrigger(
+                    task_id=self.task_id,
+                    fivetran_conn_id=self.fivetran_conn_id,
+                    connector_id=self.connector_id,
+                    poke_interval=self.poll_frequency,
+                    reschedule_wait_time=self.reschedule_wait_time,
+                ),
+                method_name="execute_complete",
+            )
```
Comment (Lee-W) [inline fivetran_provider_async/operators.py:87]: Are we going to add sync logic?

**Review by dstandish — APPROVED** [review]

File: fivetran_provider_async/operators.py (hunk 1 above)
Comment (sunank200) [reply fivetran_provider_async/operators.py:87]: i think the current implementation only has async logic

**Review by sunank200 — COMMENTED** [review]
Can we add the tests for the same?

**Review by phanikumv — COMMENTED** [review]
Need to wait for the job to complete in sync mode.I think we are just submitting the job and exiting.

**Comment by jlaneve** [comment]
yeah, going to wait for https://github.com/astronomer/airflow-provider-fivetran-async/pull/33 to finish before updating this - having the sync code in here will be helpful

**Comment by phanikumv** [comment]
> yeah, going to wait for #33 to finish before updating this - having the sync code in here will be helpful

> 

#33 has been merged

**Comment by pankajastro** [comment]
this should also be handle by https://github.com/astronomer/airflow-provider-fivetran-async/pull/40

**Pull request closed without merging**